File: backends/ts-hono/src/domain/staff/entity.ts
/**
 * スタッフドメイン エンティティモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** スタッフのドメインエンティティ。 */
export interface Staff {
  id: number;
  name: string;
  email: string;
  provider: number;
  providerId: string;
  avatar: string | null;
  role: number | null;
  lastLoginAt: Date | null;
  createdAt: Date | null;
  createdBy: number | null;
  updatedAt: Date | null;
  updatedBy: number | null;
  deletedAt: Date | null;
  version: number;
}

File: backends/ruby-rails/app/domain/staff/entity.rb
# frozen_string_literal: true
#
# スタッフエンティティを表すドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Staff
    # スタッフのエンティティです。
    Entity = Struct.new(
      :id, :name, :email, :provider, :provider_id, :avatar, :role, :last_login_at,
      :created_at, :created_by, :updated_at, :updated_by,
      :deleted_at, :deleted_by, :version,
      keyword_init: true
    )
  end
end
